### DISCUSSION TITLE
Possible to find out how many reads used to create each contig?

### DISCUSSION THREAD

**Discussion by kdbchau** [ablab/spades#837] [Q&A]
Wondering if there is an output file or a way to determine how many reads were used to generate each contig. I would need this type of information for relative abundance information for my work. Likewise, I would need to know if reads can be used multiple times in the creation of different contigs? Is something like this kept in a report anywhere or is it completely irrelevant?

After blasting the contigs, I calculate relative abundance of groups of species (i.e. families) which would be the total number of hits (typically READS not contigs) that match to a family in a sample, divided by total number of reads in a sample. Hence why I want to know how many reads are used in each contig. Hope this made sense.

**Comment by asl** [comment]
There is no such information. The connection with the reads is lost quite early. So I would suggest you to use your favourite short read mapper to align reads back to the contigs. And yes, some reads might "occur" in contigs multiple times. Think about repeats.

**Reply by kdbchau** [reply]
Just to be sure I understand, a read can be used in the creation of multiple contigs (as you mentioned, if it is a repeat) but there is no way to really grab this information from any of the output files? When using a mapper like bwa, this would undoubtedly allow reads that were not used in the generation of a contig to possibly still map to it with relatively good quality or not possible? In my understanding, this seems confounding by realigning the reads back to the contigs. Can you clarify if this might be the case? Thanks so much for the help.

**Comment by asl** [comment]
> but there is no way to really grab this information from any of the output files?

This is correct. de Bruijn graph assemblers do not assemble "reads", they assemble k-mers and there is no way to track back to reads

> his would undoubtedly allow reads that were not used in the generation of a contig to possibly still map to it with relatively good quality or not possible?

Correct. Think about read with sequencing error and inexact repeat.
